src/observer/inc.py

```diff
@@ -0,0 +1,41 @@
+import boto3
+import logging
+
+# tell boto to pipe down
+logging.getLogger('botocore').setLevel(logging.WARN)
+logging.getLogger('boto3').setLevel(logging.WARN)
+
+LOG = logging.getLogger(__name__)
+
+# TODO: cache?
+def queue(name):
+    "returns a connection to a named queue"
+    return boto3.resource('sqs').get_queue_by_name(QueueName=name)
+
+def poll(queue_obj):
+    """an infinite poll on the given queue object.
+    blocks for 20 seconds before connection is dropped and re-established"""
+    while True:
+
```

Comment:
I would suggest to insert similar lines two https://github.com/elifesciences/bot-lax-adaptor/blob/develop/src/sqs_adaptor.py#L37 to deal with stopping the process without interrupting the SQS blocking wait, which leads to a message being assigned but ignored and retried later